src-tauri/src/articulation_point.rs

```diff
@@ -0,0 +1,164 @@
+
+    for node in graph.get_nodes() {
+        if !visited.get(&node.name) {
+            articulation_point_helper(
+                &graph,
+                node.name,
+                &mut visited,
+                &mut disc,
+                &mut low,
+                &mut time,
+                parent.clone(),
+                &mut ap,
+            );
+        }
+    }
+
+    let mut articulation_points = Vec::new();
+    for key in ap.keys() {
+        if *ap.get(key) {
+            articulation_points.push(graph.get_node_idx(key.to_string()));
+        }
+    }
+    return articulation_points;
+}
+
+// Create a unit test for the Graph struct
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn initialize_graph() {
+        // Create a graph
+        let mut G = Graph::new();
+
+        // Add nodes
+        let u: String = "u".to_string();
+        let v: String = "v".to_string();
+        let w: String = "w".to_string();
+        let x: String = "x".to_string();
+        let y: String = "y".to_string();
+
+        G.add_node(u.clone());
+        G.add_node(v.clone());
+        G.add_node(w.clone());
+        G.add_node(x.clone());
+        G.add_node(y.clone());
+
+        // Add edges
+        G.add_edge(u.clone(), v.clone(), 1.0);
+        G.add_edge(u.clone(), w.clone(), 1.0);
+        G.add_edge(u.clone(), x.clone(), 1.0);
+        G.add_edge(w.clone(), x.clone(), 1.0);
+        G.add_edge(v.clone(), y.clone(), 1.0);
+        G.add_edge(x.clone(), y.clone(), 1.0);
+
+        println!("\n");
+
+        // Test the articulation point function
+        let articulation_points = articulation_point(G.clone());
+        let len_articulation_points = articulation_points.len();
+        for node in articulation_points {
+            let node = G.g.node_weight(node).unwrap();
+            println!("Articulation Point: {}", node.name);
+        }
+        assert_eq!(len_articulation_points, 0);
+
+        let mut G_2 = Graph::new();
+        let v1: String = "1".to_string();
+        let v2: String = "2".to_string();
+        let v3: String = "3".to_string();
+        let v4: String = "4".to_string();
+        let v5: String = "5".to_string();
+
+        G_2.add_node(v1.clone());
+        G_2.add_node(v2.clone());
+        G_2.add_node(v3.clone());
+        G_2.add_node(v4.clone());
+        G_2.add_node(v5.clone());
+
+        G_2.add_edge(v2.clone(), v1.clone(), 1.0);
+        G_2.add_edge(v1.clone(), v3.clone(), 1.0);
+        G_2.add_edge(v3.clone(), v2.clone(), 1.0);
+        G_2.add_edge(v1.clone(), v4.clone(), 1.0);
+        G_2.add_edge(v4.clone(), v5.clone(), 1.0);
+
+        let articulation_points = articulation_point(G_2.clone());
+        let len_articulation_points = articulation_points.len();
+        for node in articulation_points {
+            let node = G_2.g.node_weight(node).unwrap();
+            println!("Articulation Point: {}", node.name);
+        }
+
+        assert_eq!(len_articulation_points, 2);
```

Comment:
Nit: might be worth testing the `name` property of the articulation points in addition to the length of the vector